webapp/tests/test_tags.py

```diff
@@ -170,6 +170,23 @@ def test_local_tagdb(self):
   def test_redis_tagdb(self):
     return self._test_tagdb(RedisTagDB())
 
+  @patch('graphite.util.log.info')
+  @patch('graphite.tags.base.cache.get')
+  def test_tagdb_cached(self, cache_get, log_info):
+    cache_get.return_value = []
+
+    result = LocalDatabaseTagDB().find_series(['tag2=value2', 'tag1=value1'])
+
+    self.assertEqual(cache_get.call_count, 1)
+    self.assertEqual(cache_get.call_args[0][0], 'TagDB.find_series:tag1=value1:tag2=value2')
+    self.assertEqual(result, [])
+
+    self.assertEqual(log_info.call_count, 1)
+    self.assertRegexpMatches(
+      log_info.call_args[0][0],
+      'graphite\.tags\.localdatabase\.LocalDatabaseTagDB\.find_series :: completed \(cached\) in [-.e0-9]+s'
+    )
+
```

Comment:
This test just verifies the TagDB behavior when there is a cached result for a query.